### ISSUE TITLE
Alerts ".all" property method should work with stream too.

### ISSUE THREAD

**Issue by AAbouZaid** [influxdata/kapacitor#1860]
Since window node has `fillPeriod` which works as some sort of buffing, so ".all()" property method in alert node should work with stream too.
Thanks

**Comment by nathanielc** [comment]
@AAbouZaid I am not sure how `.fillPeriod` relates to the `.all` for an alert node. Can you give a complete example of what you are looking for?

**Comment by AAbouZaid** [comment]
@nathanielc Then let me rephrase the question :-)
Why stream doesn't support `.all`?
> Indicates an alert should trigger only if all points in a batch match the criteria. Does not apply to stream alerts.

**Comment by nathanielc** [comment]
A stream is unbounded, what does `all` mean in the context of a stream?

**Comment by AAbouZaid** [comment]
Using `window` with `fillPeriod` should fit that case where `all` could be
used?

**Comment by nathanielc** [comment]
Ah, I think I understand the confusion now. When the docs say that `.all` does not apply to stream alerts, it means streamed data, not alerts that run in stream mode. This is a common point of confusion with Kapacitor, but stream and batch have two contexts:

* steam vs batch tasks - this effects how the task runs, i.e query the data or listen for in on the write API
* stream vs batch data - this is just about batching data into buckets. A `|window()` node will convert stream data into batch data. This work in both batch and stream tasks.

So this is valid and works:

```
stream
     |from()
        .measurement('m')
     |window()
         .period(1m)
         .fillPeriod()
         .every(1m)
     |alert()
          .all()
          //....
```

**Comment by AAbouZaid** [comment]
Aha, I got it now.
Then this issue could be closed, and if you see it's better to clarify that in th documentation to avoid this confusion. I can make a PR for it.
